service/hook/endpoint.go

```diff
@@ -194,7 +195,23 @@ func (c *Client) HTTPHandler(w http.ResponseWriter, r *http.Request) {
 		})
 
 		if err != nil {
-			logger.Error("Failed to gather metrics from the webhook", zap.Error(err))
+			knownErrors := []string{
+				"unknown X-Github-Event in message",
+				"payload signature check failed",
+			}
+			isKnownError := false
+
+			for _, knownError := range knownErrors {
+				if strings.Contains(err.Error(), knownError) {
```

Comment:
errors.Is also an alternative, did you checked that? Also the github client does not returns typed errors?